src/Bicep.Core/Analyzers/Linter/Rules/NoHardcodedEnvironmentUrlsRule.cs

```diff
@@ -102,12 +102,14 @@ private sealed class Visitor : AstVisitor
             private readonly ImmutableArray<string> disallowedHosts;
             private readonly int minHostLen;
             private readonly ImmutableArray<string> excludedHosts;
+            private readonly ISyntaxHierarchy syntaxHierarchy;
 
-            public Visitor(ImmutableArray<string> disallowedHosts, int minHostLen, ImmutableArray<string> excludedHosts)
+            public Visitor(ImmutableArray<string> disallowedHosts, int minHostLen, ImmutableArray<string> excludedHosts, ISyntaxHierarchy syntaxHierarchy)
```

Comment:
Should we also fix this for the `metadata` syntax? (`VisitMetadataDeclarationSyntax`)

For example:
```bicep
metadata foo = '''
Should we allow management.azure.com here?
'''
```